src/pudl/transform/eia.py

```diff
@@ -1170,6 +1170,16 @@ def transform(
         "boilers_annual_eia",
     )
 
+    # Remove fields that came from input data but aren't in the
+    # corresponding SQLite tables. The data may still exist but has been
+    # moved elsewhere.
+    eia_transformed_dfs["generation_fuel_eia923"] = eia_transformed_dfs[
+        "generation_fuel_eia923"
+    ].drop(columns=["utility_name_eia"])
+    eia_transformed_dfs["generation_fuel_nuclear_eia923"] = eia_transformed_dfs[
+        "generation_fuel_nuclear_eia923"
+    ].drop(columns=["utility_name_eia"])
+
```

Comment:
I'm a little surprised that this is necessary. I thought that by default the harvesting process dropped any column that was harvested from the table it was found in (unless we specifically tell it not to rip the column out). But I could be wrong about this @cmgosnell would probably know better.

One option to deal with the issue of leftover columns that need to be harvested, but shouldn't stick around in a generic way is to use the `pudl.metadata.classes.Resource.format_df()` method, as we are in `pudl.transform.classes.AbstractTableTransformer.enforce_schema()`.

In fact, copying `enforce_schema()` almost verbatim into a new function might be the way to go. I think the only big difference is you'd need to look up the appropriate resource using the table name that is the key in that dictionary of dataframes, rather than having `self.table_id` to work with.

Eventually the plan is to refactor the EIA transforms to use the same class design that we've applied to FERC 1, but for now just iterating over all the finished dataframes at the very end of the EIA transformation process would avoid the need to special case out which columns should be present or not, and do some other basic checks that will cause the DB to complain (no null PKs, unique PKs, etc).